engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoFile.java

```diff
@@ -45,17 +49,58 @@ public EnsoFile(TruffleFile truffleFile) {
   @Builtin.Method(name = "output_stream_builtin")
   @Builtin.WrapException(from = IOException.class)
   @Builtin.ReturningGuestObject
+  @Builtin.Specialize
   @CompilerDirectives.TruffleBoundary
-  public OutputStream outputStream(OpenOption[] opts) throws IOException {
-    return this.truffleFile.newOutputStream(opts);
+  public OutputStream outputStream(Object opts,
+      EnsoContext ctx,
+      @CachedLibrary(limit = "5") InteropLibrary interop) throws IOException {
+    OpenOption[] openOptions = convertInteropArray(opts, interop, ctx, OpenOption[]::new);
+    return this.truffleFile.newOutputStream(openOptions);
   }
 
   @Builtin.Method
   @Builtin.WrapException(from = IOException.class)
+  @Builtin.Specialize
   @Builtin.ReturningGuestObject
   @CompilerDirectives.TruffleBoundary
-  public InputStream inputStream(OpenOption[] opts) throws IOException {
-    return this.truffleFile.newInputStream(opts);
+  public InputStream inputStream(Object opts,
+      EnsoContext ctx,
+      @CachedLibrary(limit = "5") InteropLibrary interop) throws IOException {
+    OpenOption[] openOptions = convertInteropArray(opts, interop, ctx, OpenOption[]::new);
+    return this.truffleFile.newInputStream(openOptions);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <T> T[] convertInteropArray(Object arr,
+      InteropLibrary interop,
+      EnsoContext ctx,
+      IntFunction<T[]> hostArrayCtor) {
+    if (!interop.hasArrayElements(arr)) {
+      var vecType = ctx.getBuiltins().vector().getType();
+      var typeError = ctx.getBuiltins().error().makeTypeError(vecType, arr, "opts");
+      throw new PanicException(typeError, interop);
+    }
+    T[] hostArr;
+    try {
+      int size = Math.toIntExact(interop.getArraySize(arr));
+      hostArr = hostArrayCtor.apply(size);
+      for (int i = 0; i < size; i++) {
+        Object elem = interop.readArrayElement(arr, i);
+        if (!ctx.getEnvironment().isHostObject(elem)) {
+          var err = ctx.getBuiltins().error().makeUnsupportedArgumentsError(
+              new Object[]{arr},
+              "Arguments to opts should be host objects from java.io package"
+          );
```

Comment:
I think this is better as `Unsupported_Argument_Types` because it has a message field, as opposed to `Type_Error` that only has expected and actual types.